plutus-benchmark/flat/Main.hs

```diff
@@ -0,0 +1,86 @@
+{-# LANGUAGE BangPatterns      #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module Main where
+
+import           Control.DeepSeq              (force)
+import           Criterion.Main
+import           Criterion.Types
+import qualified Data.ByteString              as BS
+import           Data.Map.Strict              (Map, empty, insert, lookup)
+import           Data.Maybe                   (fromJust)
+import           Data.Text                    (Text, unpack)
+import           System.Mem                   (performMajorGC)
+
+import           Language.PlutusCore          (Name (..))
+import           Language.PlutusCore.Universe
+import           Language.UntypedPlutusCore
+
+import           Codec
+import           Dataset
+import           Report
+
+import           Prelude                      hiding (lookup)
+
+type Tm = Term Name DefaultUni ()
+
+serializedContracts :: Map (Text, Text) BS.ByteString
+serializedContracts =
+  foldl update empty [ (contract, codec) | contract <- contracts, codec <- codecs ]
+  where
+    update :: Map (Text, Text) BS.ByteString
+           -> ((Text, Tm), (Text, Codec Tm))
+           -> Map (Text, Text) BS.ByteString
+    update cs ((contractName, contract), (codecName, codec)) =
+      insert (contractName, codecName) ((Codec.serialize codec) contract) cs
```

Comment:
I think you could do this with something like:

```
serializedContracts = mconcat $ do
  (contractName, contract) <- codecs
  (codecName, codec) <- contracts
  pure $ Map.singleton (contractName, codecName) ((Codec.serialize codec) contract)
```

...which to my mind is clearer. Or even...

```
serializedContracts = Map.fromList $ do
  (contractName, contract) <- codecs
  (codecName, codec) <- contracts
  [(contractName, codecName) ((Codec.serialize codec) contract)]
```

Reply:
I used the second version, and used `pure $ ...` rather than the list notation.